sleap/gui/web.py

```diff
@@ -146,6 +149,49 @@ def get_release(self, version: str) -> Release:
         )
 
 
+@attr.s(auto_attribs=True)
+class AnnouncementChecker:
+    """Checker for new announcements on the bulletin page of sleap."""
+
+    bulletin_json_path: str = BULLETIN_JSON
+    previous_announcement_date: str = prefs.prefs["announcement last seen date"]
+
+    def check_for_announcement(self) -> bool:
+        """Returns if new announcements are available."""
+        try:
+            # Attempt to open the file in read mode
+            with open(self.bulletin_json_path, 'r', encoding='utf-8') as jsf:
+                # Load the JSON data into a Python data structure
+                data = json.load(jsf)
+                latest_data = data[0]
+                
+                if latest_data['date'] != self.previous_announcement_date:
+                    return True
+        except FileNotFoundError:
+            return False
+
+    def get_latest_announcement(self) -> Optional[Tuple[str, str]]:
+        """Return latest announcements on the releases page not seen by user."""
+        success = self.check_for_announcement()
+        if success:
+            # Attempt to open the file in read mode
+            with open(self.bulletin_json_path, 'r', encoding='utf-8') as jsf:
+                # Load the JSON data into a Python data structure
+                data = json.load(jsf)
+                latest_data = data[0]
+                
+                if latest_data['date'] != self.previous_announcement_date:
+                    return (latest_data['date'], latest_data['content'])
+        return None
+
+    def update_announcement(self):
+        """Update the last seen date of announcement in preferences."""
+        announcement = self.get_latest_announcement()
+        if announcement is not None:
+            prefs.prefs["announcement last seen date"] = announcement[0]
+            prefs.prefs["announcement"] = announcement[1]
```

Comment:
The `prefs.prefs` variable should only be read in at opening and written to at closing the program. For intermediate states, we use `MainWindow.state`. The `prefs.prefs` variable stores the user preferences which are written to a file to be remembered each time the GUI is opened/closed.

Reply:
Can we directly use `MainWindow.state["announcement last seen date"]` or do we have to use it in some other way?